src/Cli/dotnet/commands/dotnet-help/HelpCommand.cs

```diff
@@ -46,75 +46,73 @@ public static void PrintVersionHeader()
-                               File.Exists(@"/sbin/xdg-open") ? @"/sbin/xdg-open" :
-                               "xdg-open";
-                psInfo = new ProcessStartInfo
-                {
-                    FileName = fileName,
-                    Arguments = docUrl
-                };
+                FileName = docUrl,
+                UseShellExecute = true
             }
-
-            return new Process
-            {
-                StartInfo = psInfo
             };
         }
 
+
+        /// <summary>
+        /// Opens the HTML help for one of our <see cref="DocumentedCommand"/>s. If the command requested isn't a
+        /// <see cref="DocumentedCommand"/>, or if it doesn't have a <see cref="DocumentedCommand.DocsLink" /> we'll just call <see cref="CliCommand.ShowHelp"/> on it.
+        /// </summary>
+        /// <returns></returns>
         public int Execute()
         {
-            if (TryGetDocsLink(
-                _parseResult.GetValue(HelpCommandParser.Argument),
-                out var docsLink) &&
-                !string.IsNullOrEmpty(docsLink))
+            var commandAndArgs = _parseResult.GetValue(HelpCommandParser.Arguments);
+            var innerParseResult = Cli.Parser.Instance.Parse(["dotnet", .. commandAndArgs]);
+            if (innerParseResult.CommandResult is { } commandResult)
             {
-                var process = ConfigureProcess(docsLink);
-                process.Start();
-                process.WaitForExit();
-                return 0;
+
+                if (commandResult.Command is DocumentedCommand documentedCommand)
+                {
+                    if (!string.IsNullOrEmpty(documentedCommand.DocsLink))
+                    {
+                        var process = ConfigureProcess(documentedCommand.DocsLink);
+                        try
+                        {
+                            process.Start();
+                            process.WaitForExit();
+                            if (process.ExitCode != 0)
+                            {
+                                // if the 'open browser' process fails, we should fallback to
+                                // calling `--help` for that command and outputting that
+                                return innerParseResult.ShowHelp();
+                            }
+                            else
+                            {
+                                return process.ExitCode;
+                            }
+                        }
+                        catch 
+                        {
+                            return innerParseResult.ShowHelp();
+                        }
+                    }
+                    else
+                    {
+                        return innerParseResult.ShowHelp();
+                    }
+                }
+                else
+                {
+                    return innerParseResult.ShowHelp();
+                }
```

Comment:
Code path simplification suggestion. Also, the `catch` should really have some kind of exception type with it. Catching everything is generally not recommended.

```suggestion
                if (commandResult.Command is DocumentedCommand documentedCommand
                    && !string.IsNullOrEmpty(documentedCommand.DocsLink))
                {
                    var process = ConfigureProcess(documentedCommand.DocsLink);
                    try
                    {
                        process.Start();
                        process.WaitForExit();
                        if (process.ExitCode == 0)
                        {
                            return process.ExitCode;
                        }
                    }
                    catch { }
                }

                // if the 'open browser' process fails, we should fallback to
                // calling `--help` for that command and outputting that
                return innerParseResult.ShowHelp();
```

Reply:
Valid point on the `catch` - but two points that led me to not care as much:
* this is deliberately a fall-back - if _anything_ went wrong launching the browser, then we should just emit the CLI help. We're not doing any diagnostics or trying to recover here at all.
* for urls, the most common exception I see happening here is on Linux the various shell-opening utilities may not be installed. That'll be raised as an exception by the runtime, and that's the only exceptional case I'd ever really expect to see here.